learn/work/fairness-readmission/index.qmd

```diff
@@ -0,0 +1,517 @@
+set.seed(1)
+wflow_set_fit <- 
+  workflow_map(
+    wflow_set, 
+    verbose = TRUE, 
+    seed = 1, 
+    metrics = m_set,
+    resamples = readmission_folds
+  )
+```
+
+```{r}
+#| label: "save-wflow-set"
+#| eval: !expr eval_fits
+#| include: false
+save(wflow_set_fit, file = saved_fits)
+```
+
+A fitted workflow set looks just like the unfitted workflow set we saw previously, except that information on the tuning process is now stored in the `option` and `result` variables for each modeling workflow:
+
+```{r}
+#| label: "print-wflow-set"
+wflow_set_fit
+```
+
+## Model Selection
+
+Now that we've evaluated a number of models with a variety of metrics, we can explore the results to determine our optimal model. Beginning by a quick exploratory plot of our metrics:
+
+<!-- TODO: i'd really like to use `autoplot(wflow_set_fit, rank_metric = "roc_auc")` here, but the plot isn't usable without differentiating between the recipes. -->
+
+```{r}
+#| label: "plot-wflow-set-fit"
+wflow_set_fit %>%
+  rowwise() %>%
+  mutate(metrics = list(collect_metrics(result))) %>%
+  select(wflow_id, metrics) %>%
+  unnest(metrics) %>%
+  mutate(
+    .,
+    .config = paste0(wflow_id, .config)
+  ) %>%
+  mutate(
+    .config = factor(.config, 
+                     levels = filter(., .metric == "roc_auc") %>%
+                              arrange(desc(mean)) %>%
+                              pull(.config) %>% 
+                              unique()
+                     )
+  ) %>%
+  ggplot(aes(x = .config, y = mean, color = wflow_id)) +
+  facet_wrap(vars(.metric), scales = "free") +
+  geom_point() +
+  scale_x_discrete(breaks = NULL)
+```
+
+Each dot in the plots above represents one configuration of a modeling workflow. For each modeling workflow besides logistic regressions, we've tried 10 unique combinations of hyperparameters. For the workflows using logistic regressions, we've tried only one, since we didn't mark any parameters for tuning using the `tune()` placeholder. The dots are "ranked" by `roc_auc()` in that, in each pane, the left-most models took the highest `roc_auc()` values.
+
+The fairness metrics `demographic_parity()`, `equal_opportunity()`, and `equalized_odds()` all take values very close to zero for many models. Notably, the left-most models (i.e. the most performant models with respect to `roc_auc()`) happen to be among the most fair according to the metrics we've chosen.
+
+More concretely, we can rank the model configurations to examine only the most performant models:
+
+```{r}
+#| label: "rank-results"
+rank_results(wflow_set_fit, rank_metric = "roc_auc") %>%
+  filter(.metric == "roc_auc")
+```
+
+As in the plot above, we see that almost all of the most performant model configurations arise from the boosted tree modeling workflow. Let's examine the results for specifically the modeling workflow that encodes age as a number more thoroughly:
+
+```{r}
+#| label: "autoplot-age-bt"
+autoplot(wflow_set_fit, id = "age_bt")
```

Comment:
Can you set fig width and height to make the metric labels more readable?

Reply:
Yes! 5dce621. Let me know if that does the trick---the metric names are now fully visible but I'm curious how the size looks on others' setups.

Reply:
Wrapped my head around the sizing a bit better--91ca48a.